src/Agent.Plugins/PipelineArtifact/BuildServer.cs

```diff
@@ -54,5 +54,15 @@ public BuildServer(VssConnection connection)
         {
             return await _buildHttpClient.GetArtifactAsync(projectId, buildId, name, cancellationToken: cancellationToken);
         }
+
+        //Get artifact with project name.
+        public async Task<BuildArtifact> GetArtifactWithProjectAsync(
+            string project,
+            int buildId,
+            string name,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await _buildHttpClient.GetArtifactAsync(project, buildId, name, cancellationToken: cancellationToken);
```

Comment:
await not necessary just return the task and make it non asyc

Reply:
Why wouldn't you await?

Reply:
You can simplify the function by returning just the Task and having the caller await.

Reply:
Not sure that's correct. Can you provide an example? Maybe I am misunderstanding. You would still await in the caller. It's async all the way up and down.